### PR TITLE
Feature/93 nft activity log

### LINKED ISSUE #93: NFT transaction history
Only swap transactions needs to be logged.

Image

### PR DESCRIPTION
**Pull request by piersss**
# Description

Fixes #93 

Tested with both ERC721 and ERC1155 on goerli.

CrypToadz ERC721 0xD000F000Aa1F8accbd5815056Ea32A54777b2Fc4:
http://localhost:3000/#/nft-detail/220

Screenshot 2023-08-06 at 16 05 55

EldaruneGameCollection ERC1155 0x51d54dc3b9ebdD107660055841334893370a1a5b:
http://localhost:3000/#/nft-detail/10002

Screenshot 2023-08-06 at 16 19 18

# Steps to test the change

Help the reviewer test what you implemented/changed. Include all the steps needed to perform the test.

Example:

1. Set REACT_APP_COLLECTION_TOKEN to 0xD000F000Aa1F8accbd5815056Ea32A54777b2Fc4, 0x51d54dc3b9ebdD107660055841334893370a1a5b or any other ERC721 / ERC1155 token.
2. Go to a random tokenId page like `nft-detail/1`.

# Checklist:

- [x] I have read and understood [CONTRIBUTING.MD](https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md)
- [x] I have included the ticket number in my PR
- [x] I have performed a self-review of my own code
- [x] I have commented the parts where my code is hard to understand

### CODE REVIEW

File: src/redux/stores/nftDetail/nftDetailHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+export const getErc1155Logs = async (
+  chainId: number,
+  collectionToken: string,
+  provider: Web3Provider,
+  tokenId: number,
+): Promise<NftTransactionLog[]> => {
+  const contract = new ethers.Contract(collectionToken, erc1155Contract.abi, provider);
+  const transferFilter = contract.filters.TransferSingle(null, null, null, null);
```
Comment (piersss) [inline src/redux/stores/nftDetail/nftDetailHelpers.ts:43]: @gpxl-dev Any idea why `contract.filters.TransferSingle(null, null, null, tokenId);` doesn't work?

```
Error: cannot filter non-indexed parameters; must be null (argument="contract.id", value="10055", code=INVALID_ARGUMENT, version=abi/5.7.0)
```

https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#IERC1155-TransferSingle-address-address-address-uint256-uint256-

Comment (gpxl-dev) [reply src/redux/stores/nftDetail/nftDetailHelpers.ts:43]: `event TransferSingle(address indexed _operator, address indexed _from, address indexed _to, uint256 _id, uint256 _value);`

Yes - as the error suggests the parameter you are trying to filter by (`_id` in the signature above) is not indexed. You can only filer by indexed params.

So in this case you would need to get all of the transfer events and manually filter them. It sucks, I know...

File: src/redux/stores/config/configHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { TokenKinds } from '@airswap/constants';
+import erc20AbiContract from '@openzeppelin/contracts/build/contracts/ERC20.json';
+import erc721AbiContract from '@openzeppelin/contracts/build/contracts/ERC721.json';
+import { ethers } from 'ethers';
+
+export const getCollectionTokenTokenKind = async (provider: ethers.providers.Web3Provider, address: string): Promise<TokenKinds | undefined> => {
```
Comment (personnamedmike) [inline src/redux/stores/config/configHelpers.ts:6]: can probably rename this from `getCollectionTokenTokenKind` to `getCollectionTokenKind`

File: src/redux/stores/config/configHooks.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  } = useAppSelector(state => state.config);
+
+  useEffect(() => {
+    if (!library || chainId !== configChainId) {
+      return;
+    }
+
+    dispatch(getCollectionTokenKind({ address: collectionToken, provider: library }));
+    dispatch(getCurrencyTokenKind({ address: currencyToken, provider: library }));
+  }, [chainId]);
```
Comment (personnamedmike) [inline src/redux/stores/config/configHooks.ts:30]: can add `library` and `configChainId` to dependency array

File: src/redux/stores/config/configHooks.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    ) {
+      dispatch(setIsSuccessful(true));
+    }
+  }, [collectionTokenKind]);
+
+  useEffect(() => {
+    if (chainId !== configChainId) {
+      dispatch(reset());
+    }
+  }, [chainId]);
```
Comment (personnamedmike) [inline src/redux/stores/config/configHooks.ts:47]: can add `configChainId` to dep array`

File: src/redux/stores/config/configSlice.ts [hunk 5]
```diff
@@ -1,14 +1,27 @@
...
+  isLoadingCurrencyTokenKind: boolean;
+  isSuccessful: boolean;
   chainId: number;
   currencyToken: string;
+  currencyTokenKind?: TokenKinds;
   collectionToken: string;
+  collectionTokenKind?: TokenKinds;
   collectionName: string;
   collectionImage: string;
   ipfcGatewayURL: string;
```
Comment (personnamedmike) [inline src/redux/stores/config/configSlice.ts:18]: is this supposed to be `ipfsGatewayURL`? Currently it's ipfc

File: src/redux/stores/config/configHooks.ts (hunk 3 above)
Comment (piersss) [reply src/redux/stores/config/configHooks.ts:30]: This will add unintended behaviours.

File: src/redux/stores/config/configHooks.ts (hunk 4 above)
Comment (piersss) [reply src/redux/stores/config/configHooks.ts:47]: Same, it wouldn't improve anything.

**Review by personnamedmike — APPROVED** [review]

**Pull request merged**